src/ir/auth_logic/universe_relation_insertion.h

```diff
@@ -0,0 +1,449 @@
+/*
```

Comment:
There is way too much stuff in this header file. While it is often preferable to put the definitions of small methods directly in the header, it should usually be the case that someone can pop open a header and scan it to get an idea of what the class does. Once the details become complex, they should be moved to an associated `.cc` file.

In particular, this file contains a number of `static` methods which do not take advantage of special access to private members of the class. That's a warning sign of bad factoring: maybe those shouldn't be members of the class at all, or should be helpers in an unnamed namespace in the `.cc` file.

Inner classes in a header file are similarly questionable. When they are exposed to clients, it may be right to put them in the header. Same for if they are very small and putting them in the header prevents from having to create a `.cc` file. Otherwise, maybe they should be classes in an anonymous namespace in the `.cc` file or should be broken out into their own header.

Reply:
fixed